### ISSUE TITLE
apm not sending any metrics 

### ISSUE THREAD

**Issue by roeehershko** [elastic/apm-agent-nodejs#689]
**Describe the bug**

**To Reproduce**
copy paste the example configuration:
```
// Add this to the VERY top of the first file loaded in your app
var apm = require('elastic-apm-node').start({
    // Override service name from package.json
    // Allowed characters: a-z, A-Z, 0-9, -, _, and space
    serviceName: 'test',


    // Set custom APM Server URL (default: http://localhost:8200)
    serverUrl: 'http://x.x.x.x:8200',

    logLevel: 'debug',
});

var app = require('express')()

app.get('/', function (req, res) {
    res.send('Hello World!')

})

app.listen(3001);

console.log('Listening!');
```

**Expected behavior**
Metrics should be available on kibana,
i use the same apm-server for anther php project which works and sends metrics

**Environment (please complete the following information)**
- OS: [e.g. Linux]
- Node.js version: 10
- APM Server version: latest
- Agent version: latest

**How are you starting the agent? (please tick one of the boxes)**
- [*] Calling `agent.start()` directly (e.g. `require('elastic-apm-node').start(...)`)
- [ ] Requiring `elastic-apm-node/start` from within the source code
- [ ] Starting node with `-r elastic-apm-node/start`

**Additional context**
This is my console output for starting the server & doing 2 requests to the same URL (/):
```
# Starting NodeJS server (node index.js)
adding hook to Node.js module loader
shimming finalhandler@1.1.1 module
shimming http@10.9.0 module
shimming http.Server.prototype.emit function
shimming http.request function
shimming http.ServerResponse.prototype.writeHead function
shimming express@4.16.4 module
shimming express.Router.process_params function
shimming express.Router.use function
shimming express.static function
copying property mime from express.static
Listening!

# Sending request 1
intercepted request event call to http.Server.prototype.emit
start trace { trans: 'a4353a4cb368462f',
  parent: undefined,
  trace: 'eee240d9d7e6ca2d6162a04a71b4b163',
  name: null,
  type: null }
shimming express.Router.Layer.handle function
shimming express.Router.Layer.handle function
shimming express.Router.Layer.handle function
setting transaction result { trans: 'a4353a4cb368462f',
  parent: undefined,
  trace: 'eee240d9d7e6ca2d6162a04a71b4b163',
  result: 'HTTP 3xx' }
setting default transaction name: GET / { trans: 'a4353a4cb368462f',
  parent: undefined,
  trace: 'eee240d9d7e6ca2d6162a04a71b4b163' }
sending transaction { trans: 'a4353a4cb368462f',
  trace: 'eee240d9d7e6ca2d6162a04a71b4b163' }
ended transaction { trans: 'a4353a4cb368462f',
  parent: undefined,
  trace: 'eee240d9d7e6ca2d6162a04a71b4b163',
  type: 'request',
  result: 'HTTP 3xx',
  name: 'GET /' }
no active transaction found - cannot build new span

# Sending request 2
intercepted call to http.request { id: null }
intercepted request event call to http.Server.prototype.emit
start trace { trans: '57f81d72a475ea8f',
  parent: undefined,
  trace: '83252f5e13c73e02e1905d4e61bb41c1',
  name: null,
  type: null }
setting transaction result { trans: '57f81d72a475ea8f',
  parent: undefined,
  trace: '83252f5e13c73e02e1905d4e61bb41c1',
  result: 'HTTP 3xx' }
setting default transaction name: GET / { trans: '57f81d72a475ea8f',
  parent: undefined,
  trace: '83252f5e13c73e02e1905d4e61bb41c1' }
sending transaction { trans: '57f81d72a475ea8f',
  trace: '83252f5e13c73e02e1905d4e61bb41c1' }
ended transaction { trans: '57f81d72a475ea8f',
  parent: undefined,
  trace: '83252f5e13c73e02e1905d4e61bb41c1',
  type: 'request',
  result: 'HTTP 3xx',
  name: 'GET /' }
```
APM server logs are empty, nothing is coming

**Comment by watson** [comment]
Hi @roeehershko 

Thanks for reaching out. I just tried to run your example, but could not reproduce the bug. The log line `sending transaction` indicates that the agent have started to send the data to the APM Server - though it doesn't indicate that it was successful. It should show an error message soon after if it couldn't send the data, but I don't see that in the logs you posted.

You write that you're using the latest APM Server. Are you aware that we _just_ released APM Server version 6.5 which has a slightly different API? Only version 2.0.0+ of the agent supports that new API. If you are using an earlier version of the APM Server, please use the latest 1.x version of the agent. Could that be the issue?

**Comment by roeehershko** [comment]
Yea my issue was version after all, i didnt noticed i used 6.4 server and 6.5 APM, after upgrading the server it started to work.
you can close this, thanks!

**Comment by watson** [comment]
Thanks for the feedback. I did just discover that the new agent will not output any error message if it gets a 404 from the APM Server - which was probably what happened in your case. It was supposed to do that, but there is a bug so the error got swallowed. This is being fixed in #692 so that other people in the future will get a better experience. Sorry for the confusion.
